File: Backend/Rentify.Domain/Entities/Property.cs
namespace Rentify.Domain.Entities;

public class Property : BaseEntity
{
    public Guid Owner { get; set; }
    public string Place { get; set; } = string.Empty;
    public int NoOfBedroom { get; set; }
    public int NoOfBathroom { get; set; }
    public int SizeinSqft { get; set; }
    public bool NearbySchool { get; set; }
    public bool NearbyCollege { get; set; }
    public bool NearbyHospital { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public List<Guid> Likes { get; set; } = [];
}

File: Frontend/src/Models/property.ts
import { PropertyModel } from "./propertyModel";

// Defins the Property class
export class Property {
  constructor(
    public id: string,
    public place: string,
    public noOfBedroom: number,
    public noOfBathroom: number,
    public sizeinSqft: number,
    public nearbySchool: boolean,
    public nearbyCollege: boolean,
    public nearbyHospital: boolean,
    public description: string,
    public price: number,
    public lastUpdated: string
  ) {}

  // Convert Property to PropertyModel
  toModel(): PropertyModel {
    return new PropertyModel(
      this.id,
      this.place,
      this.noOfBedroom,
      this.noOfBathroom,
      this.sizeinSqft,
      this.nearbySchool,
      this.nearbyCollege,
      this.nearbyHospital,
      this.description,
      this.price,
      this.lastUpdated
    );
  }

  // Static method to convert a list of Property to a list of PropertyModel
  static toModelList(properties: Property[]): PropertyModel[] {
    return properties.map((property: Property) => property.toModel());
  }
}
